include/GafferScene/Shader.h

```diff
@@ -130,6 +130,9 @@ class Shader : public Gaffer::DependencyNode
 		/// May be reimplemented in derived classes to deal with special cases.
 		virtual IECore::DataPtr parameterValue( const Gaffer::Plug *parameterPlug ) const;
 
+		virtual void attributesHash( IECore::MurmurHash &h, const Gaffer::Plug *output ) const;
```

Comment:
Could you reverse the order to match the equivalent hash methods elsewhere, like so :

```virtual void attributesHash( const Gaffer::Plug *output,  IECore::MurmurHash &h ) const;```

The order is always "hash this thing into that output variable".